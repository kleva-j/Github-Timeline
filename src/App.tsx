import { Redirect, Route, Switch } from 'react-router-dom';
import { GraphQLJobs } from 'views/GraphqlJobs';
import { NotFound } from 'views/404';
import { Login } from 'views/Login';

const App: React.FC = () => (
  <Switch>
    <Redirect exact from="/" to="/jobs" />
    <Route exact path="/jobs" component={GraphQLJobs} />
    <Route exact path="/login" component={Login} />
    <Route exact path="*" component={NotFound} />
  </Switch>
);

export default App;
