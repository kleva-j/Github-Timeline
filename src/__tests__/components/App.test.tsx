import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import { render, cleanup } from '@testing-library/react';

import App from '../../App';

describe('App', () => {
  afterEach(cleanup);

  test('renders the app', () => {
    const { container, getByText } = render(
      <MockedProvider mocks={[]} addTypename={false}>
        <App />
      </MockedProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(getByText('Timeline')).toBeTruthy();
  });
});
