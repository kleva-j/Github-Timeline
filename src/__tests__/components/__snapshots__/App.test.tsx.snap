// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 404 Not Found 1`] = `
<h1>
  404 NOT FOUND
</h1>
`;

exports[`App renders the app 1`] = `
<section
  class="grid__section min-h-screen bg-primary-bg"
>
  <button
    class="sc-bdfBwQ lfJroc"
  >
    Timeline
  </button>
</section>
`;
